### PR TITLE
Tensorflow_CC 1.4

### PR DESCRIPTION
**Pull request by Chunosov**
Issue https://github.com/dividiti/ck-tensorflow-private/issues/9

This branch was incidentally derived from `tensorflow-cc-prebuilt-32` (PR https://github.com/ctuning/ck-tensorflow/pull/34) so this PR should be merged after 34.

### CODE REVIEW

File: package/lib-tensorflow_cc-shared-1.4.0/.cm/meta.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      "add_to_path": "no", 
+      "local": "no", 
+      "name": "Bazel tool", 
+      "sort": 40, 
+      "tags": "tool,bazel", 
+      "version_from": [
+        0, 
+        5
+      ], 
+      "version_to": [
```
Comment (psyhtest) [inline package/lib-tensorflow_cc-shared-1.4.0/.cm/meta.json:38]: We should be able to build at least with current Bazel 0.6 and 0.7...

File: package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+set JOB_COUNT=%CK_HOST_CPU_NUMBER_OF_PROCESSORS%
+
+set CK_CMAKE_EXTRA=%CK_CMAKE_EXTRA% ^
+ -DOPENCL_ROOT:PATH="%CK_ENV_LIB_OPENCL%" ^
+ -DOPENCL_LIBRARIES:FILEPATH="%CK_ENV_LIB_OPENCL_DYNAMIC_NAME%" ^
+ -DOPENCL_INCLUDE_DIRS:PATH="%CK_ENV_LIB_OPENCL_INCLUDE%" ^
+ -DCMAKE_EXE_LINKER_FLAGS=-L"%CK_ENV_LIB_OPENCL_LIB%" ^
+ -DTUNERS=ON ^
+ -DCLTUNE_ROOT:PATH="%CK_ENV_TOOL_CLTUNE%" ^
```
Comment (psyhtest) [inline package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat:33]: This is odd. Why are OpenCL and CLTune needed?

File: package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export JOB_COUNT=${CK_HOST_CPU_NUMBER_OF_PROCESSORS}
+
+export CK_CMAKE_EXTRA="${CK_CMAKE_EXTRA} \
+ -DOPENCL_ROOT:PATH=${CK_ENV_LIB_OPENCL} \
+ -DOPENCL_LIBRARIES:FILEPATH=${CK_ENV_LIB_OPENCL_LIB}/${CK_ENV_LIB_OPENCL_DYNAMIC_NAME} \
+ -DOPENCL_INCLUDE_DIRS:PATH=${CK_ENV_LIB_OPENCL_INCLUDE} \
+ -DTUNERS=ON \
+ -DCLTUNE_ROOT:PATH=${CK_ENV_TOOL_CLTUNE} \
+ -DCLIENTS=ON \
+ -DCBLAS_INCLUDE_DIRS:PATH=${CK_ENV_LIB_OPENBLAS_INCLUDE} \
```
Comment (psyhtest) [inline package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.sh:29]: CLBLAS?

File: package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat (hunk 2 above)
Comment (Chunosov) [reply package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat:33]: Not sure. It is the same as in package for TF 1.3. Should I check building for android?

File: package/lib-tensorflow_cc-shared-1.4.0/.cm/meta.json (hunk 1 above)
Comment (Chunosov) [reply package/lib-tensorflow_cc-shared-1.4.0/.cm/meta.json:38]: I've increased version to 0.7 but we have only bazel 0.6.1 in ck_env, so I've checked if it is built with this version only.
Should we add `package ck_env:package:tool-bazel-0.7.*-linux` now ?

Comment (psyhtest) [reply package/lib-tensorflow_cc-shared-1.4.0/.cm/meta.json:38]: > Should we add package ck_env:package:tool-bazel-0.7.*-linux now

Yes, please: https://github.com/dividiti/ck-tensorflow-private/issues/10

File: package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat (hunk 2 above)
Comment (psyhtest) [reply package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.bat:33]: Do you have a recent Android device?

File: package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.sh (hunk 3 above)
Comment (psyhtest) [reply package/lib-tensorflow_cc-shared-1.4.0/scripts.android/install.sh:29]: I think we can leave this as it is and clean up later.

**Comment by Chunosov** [comment]
I've updated `soft:lib.tensorflow_cc`, it can create additional variables when it have found new TF library `libtensorflow_framework.so` and new required include path. Test programs were updated as well. Now they can be built against both `lib-tensorflow_cc-shared-1.3.0` and `lib-tensorflow_cc-shared-1.4.0`

**Comment by psyhtest** [comment]
Thanks! I've opened #37 to clean up and test this package on Android (not urgent).

**Pull request merged**